cosmo_tester/test_suites/cli/__init__.py

```diff
@@ -57,6 +63,85 @@ def _test_cfy_install(run, example, paths, logger):
     example.check_files()
 
 
+def _set_ssh_in_profile(run, example, paths):
+    run(
+        '{cfy} profiles set --ssh-user {ssh_user} --ssh-key {ssh_key}'.format(
+            cfy=paths['cfy'],
+            ssh_user=example.manager.username,
+            ssh_key=paths['ssh_key'],
+        )
+    )
+
+
+def _test_cfy_logs(run, cli_host, example, paths, logger):
+    _set_ssh_in_profile(run, example, paths)
+
+    linebreak = '\\r\\n' if 'C:\\' in paths['cfy'] else '\n'
+    logger.info('Testing `cfy logs download`')
+    logs_dump_filepath = str(run(
+        '{cfy} logs download '.format(cfy=paths['cfy'])
+    ).stdout).split('archive to:')[1].split(linebreak)[0]
+
+    # Not working on windows!
+    local_logs_dump_filepath = os.path.join(tempfile.mkdtemp(), 'logs.tar')
+    cli_host.get_remote_file(logs_dump_filepath, local_logs_dump_filepath)
+    with closing(tarfile.open(name=local_logs_dump_filepath)) as tar:
+        files = [f.name for f in tar.getmembers()]
+        assert 'cloudify/journalctl.log' in files
+        assert 'cloudify/nginx/cloudify.access.log' in files
+
+    logger.info('Testing `cfy logs backup`')
+    run('{cfy} logs backup --verbose'.format(cfy=paths['cfy']))
+    output = example.manager.run_command('ls /var/log').stdout
+    assert 'cloudify-manager-logs_' in output
+
+    logger.info('Testing `cfy logs purge`')
+    run('{cfy} logs purge --force'.format(cfy=paths['cfy']))
+    example.manager.run_command(
+        'test ! -s /var/log/cloudify/manager/cfy_manager.log')
```

Comment:
Do we not empty any other logs when we run this?

Reply:
That's the actual code it runs:
`        conn.run(
            'for f in $(sudo find /var/log/cloudify -name "*" -type f); '
            'do sudo truncate -s 0 $f; '
            'done'
        )`

Reply:
Huh... we should correct that at some point, probably to: sudo find /var/log/cloudify -type f -exec sudo truncate -s0 {}\;

Still, not for this PR.

Reply:
So once again this one should probably be stopping services before it does the purge, then checking everything under that dir- e.g. I think this'll work (not tested):
sudo find /var/log/cloudify -type f test ! -s {} || echo {} && exit 1 \;

That should check each file under /var/log/cloudify is size zero; if not, it should echo the filename then exit nonzero, which should make the run_command fail.